concordium-node/src/bin/collector.rs

```diff
@@ -120,24 +121,25 @@ struct ConfigCli {
         env = "CONCORDIUM_NODE_COLLECTOR_MAX_GRPC_FAILURES_ALLOWED"
     )]
     pub max_grpc_failures_allowed: u64,
+    #[cfg(target_os = "macos")]
+    #[structopt(
+        long = "net-name",
+        help = "The net named used for logging on macOS, e.g. 'mainnet'. Messages will be logged \
+                with the subsystem 'software.concordium.<net-name>.node', which can be searched \
+                for in Console.app.",
+        env = "CONCORDIUM_NODE_COLLECTOR_MACOS_NET_NAME"
+    )]
+    pub net_name: String,
 }
 
 #[tokio::main]
 async fn main() {
     let conf = ConfigCli::from_args();
 
-    // Prepare the logger
-    let env = if conf.trace {
-        Env::default().filter_or("LOG_LEVEL", "trace")
-    } else if conf.debug {
-        Env::default().filter_or("LOG_LEVEL", "debug")
-    } else if conf.info {
-        Env::default().filter_or("LOG_LEVEL", "info")
-    } else {
-        Env::default().filter_or("LOG_LEVEL", "warn")
-    };
-
-    setup_logger_env(env, conf.no_log_timestamp);
+    #[cfg(target_os = "macos")]
+    setup_logger(conf.trace, conf.debug, &conf.net_name);
+    #[cfg(not(target_os = "macos"))]
+    setup_logger(conf.trace, conf.debug, conf.no_log_timestamp);
```

Comment:
I am not sure this is a good idea to only support the bespoke system logger on the mac.

I think a better solution would be to have configuration option `CONCORDIUM_NODE_COLLECTOR_USE_MAC_LOG` (or whatever name you deem good) which is an `Option<String>`. If given the system logger is used, and the provided string is the service name (`software.concordium.mainnet...`).

The same comment applies to the node configuration below.